daemon/algod/api/server/v2/test/helpers.go

```diff
@@ -77,11 +79,47 @@ var poolAddrResponseGolden = model.AccountResponse{
 	MinBalance:                  100000,
 }
 var txnPoolGolden = make([]transactions.SignedTxn, 2)
+var poolDeltaResponseGolden = model.RoundStateDelta{
```

Comment:
@Eric-Warehime I apologize for the broken build caused by https://github.com/Eric-Warehime/go-algorand/pull/2.  Here is a git patch to address https://app.circleci.com/pipelines/github/algorand/go-algorand/10824/workflows/f8ea6e8c-aede-4ae3-a54f-931594682b96/jobs/185975/parallel-runs/4?filterBy=FAILED:  [diff.txt](https://github.com/algorand/go-algorand/files/10077352/diff.txt).

I'd appreciate your review of the git patch to confirm the suggested changes are _reasonable_.  If it doesn't seem right, let me know and I will dig deeper.

Reply:
A point to consider - It's unclear to me if it's an issue?

While working on https://github.com/Eric-Warehime/go-algorand/pull/3, I realized the above git patch does _not_ round-trip identically.  Round-trip means - Deserialize from JSON-to-object -> serialize object-to-JSON -> compare both JSONs.

* In https://github.com/Eric-Warehime/go-algorand/pull/3, the JSON _before_ and _after_ blotting is different.  After blotting, the empty fields (e.g. `CreatedApps`, `AppsLocalState`) are _removed_.
* It's unclear to me why the JSON returned from server (before blotting) contains these empty fields.  These fields are marked `omitempty`.  I presume it relates to Go's interpretation of nil values vs zero length arrays for a pointer.
* I'm not sure how deep it's worth diving here.  I raise the note because the lack of identical round-tripping feels _strange_ to me.

Reply:
The reason the serialization provided by the approvals package differs from what you see output from the API is that our serialization is customized by using go-codec. Specifically the [RecursiveEmptyCheck](https://github.com/algorand/go-algorand/blob/master/protocol/codec.go#L64) which according to [the docs](https://github.com/algorand/go-codec/blob/0d0c2ea84d71dd723199617c73fa7e67a2810b7f/codec/encode.go#L159) 
> controls whether we descend into interfaces, structs and pointers when checking if a value is empty.

The effect is that pointers to empty arrays in our serialization are output as `nil` whereas the standard Json library would output the empty array.

Reply:
@Eric-Warehime Resolving - Thanks for the pointer.  

I had missed the salient point - API handlers use echo's JSON serializer rather than the go-codec JSON serializer.  In turn, it explains why the round-trip does _not_ match.